### PR TITLE
chore: add v0.3.0-rc2 upgrade handler

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/upgrades.go [hunk 1]
```diff
@@ -50,7 +51,7 @@ func (app *App) registerUpgrade0_1_4(_ upgradetypes.Plan) {
 	)
 }
 
-//nolint: all
+// nolint: all
```
Comment (adamewozniak) [inline app/upgrades.go:54]:
```suggestion
//nolint: all
```

**Review by rbajollari — APPROVED** [review]

File: app/upgrades.go [hunk 2]
```diff
@@ -150,6 +151,16 @@ func (app *App) registerUpgrade0_3_0(upgradeInfo upgradetypes.Plan) {
 	})
 }
 
+func (app *App) registerUpgrade0_3_0_rc2(upgradeInfo upgradetypes.Plan) {
```
Comment (adamewozniak) [inline app/upgrades.go:154]:
```suggestion
func (app *App) registerUpgrade0_3_0_rc2(_ upgradetypes.Plan) {
```

File: app/upgrades.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ func (app App) RegisterUpgradeHandlers() {
 	app.registerUpgrade0_2_1(upgradeInfo)
 	app.registerUpgrade0_2_2(upgradeInfo)
 	app.registerUpgrade0_3_0(upgradeInfo)
+	app.registerUpgrade0_3_0_rc2(upgradeInfo)
```
Comment (adamewozniak) [inline app/upgrades.go:38]:
```suggestion
	app.registerUpgrade0_3_0Rc2(upgradeInfo)
```

File: app/upgrades.go [hunk 4]
```diff
@@ -150,6 +151,16 @@ func (app *App) registerUpgrade0_3_0(upgradeInfo upgradetypes.Plan) {
 	})
 }
 
+func (app *App) registerUpgrade0_3_0_rc2(_ upgradetypes.Plan) {
```
Comment (adamewozniak) [inline app/upgrades.go:154]:
```suggestion
func (app *App) registerUpgrade0_3_0Rc2(_ upgradetypes.Plan) {
```

**Comment by adamewozniak** [comment]
Looks like the non-determinism test panics due to empty validators at some point with seed `1615392044997054881`
I'll re-run it and see if it fails with another seed, I doubt this PR caused that

**Pull request merged**